ai-player.js

```diff
@@ -235,10 +235,46 @@ function createAiPlayer(game, options) {
         if (state.state.name == stateNames.FINAL_ACTION_RESPONSE) {
             return false;
         }
+
```

Comment:
This function is called in two different places and I think you need to handle them differently:

- An opponent has played an action and you are deciding whether to challenge him (your code handles this case)
- You have played an action, your opponent has blocked the action, and you are deciding whether to challenge him (I think your code would make an incorrect choice in this case. It should be enough to wrap your checks with an `if (state.state.name == stateNames.ACTION_RESPONSE)`)